### PR TITLE
feat: add support for relative dates on --date argument

### LINKED ISSUE #74: Add support for relative dates in the `--date` argument
**Is your feature request related to a problem? Please describe.**
Since version 0.7, the `--day`, `--month` and `--year` arguments have been removed in the favor of a single, more simple `--date` one.
But as a side-effect, calculating the ephemerides of the next days is now much more painful, because we now need to give the entire date: before, we could give the --day only for instance, and the month and year were automatically guessed to the current ones.

**Describe the solution you'd like**
Add the capacity for the `--date` argument to understand relative dates. The solution I imagine is something inspired from [PHP's `DateInterval` constructor](https://www.php.net/manual/en/dateinterval.construct.php), but in a more user-friendly style.

The format would follow the following regular expression:
```regexp
^([+-])(([0-9]+)y)?[ ]?(([0-9]+)m)?[ ]?(([0-9]+)d)?$
```
- The string starts with a `+` or a `-` character.
- Each duration period is represented by an integer value followed by one of the following period designators:
   - `d` for days,
   - `m` for months,
   - `y` for years.
- Each part of the duration value is optional, but there must be at least one of them for the string to be a valid period.
- There may be spaces between each part of the interval.

Examples of valid intervals:
- `+1d` (tomorrow)
- `+1m2d` (in one months and two days)
- `+1y2m` (in one year and two months)
- `+1y 2m` (same)

**Describe alternatives you've considered**
Currently, the only possible alternative is to give the entire date in the `YYYY-MM-DD` format.

### PR DESCRIPTION
**Pull request by Brainface1**
| Q              | A
| -------------- | ---
| Bug fix?       | no
| New feature?   | yes
| Related issues | Fix #74
| Has BC-break   | yes
| License        | GNU AGPL-v3

**Checklist:**
- [✔️] I have updated the manpage

This PR adds support for specifying relative dates, i.e. intervals.

### CODE REVIEW

**Comment by Brainface1** [comment]
I hope i did everything correctly this time, but i am a bit confused why it fails importing the dateutil package. I updated (and pushed) the Pipfile correctly.

And thanks for helping me with everything

**Comment by Deuchnord** [comment]
Your PR is at a very better state now! :+1: 

I don't see any changes in the `Pipfile.lock`, so it has not installed it.
Btw I don't see any package named [`dateutil` on PyPi](https://pypi.org/search/?q=dateutil), can you double-check the library's name?

Once you're one, remove the line you added in the Pipfile and invoke `pipenv install <name of the package>` instead. The command will do the necessary work to update those files :)

File: kosmorrolib/main.py [hunk 1]
```diff
@@ -99,14 +100,30 @@ def main():
     return 0
 
 
-def get_date(yyyymmdd: str) -> date:
-    if not re.match(r'^\d{4}-\d{2}-\d{2}$', yyyymmdd):
-        raise ValueError(_('The date {date} does not match the required YYYY-MM-DD format.').format(date=yyyymmdd))
+def get_date(date_arg: str) -> date:
```
Comment (Deuchnord) [inline kosmorrolib/main.py:103]: Maybe we should move this function to `core.py` and add some unit tests on it… WDYT?

File: manpage/kosmorro.1.md [hunk 2]
```diff
@@ -23,7 +23,7 @@
     the observer's longitude on Earth
 
 `--date=`_DATE_, `-d` _DATE_  
-    The date for which the ephemerides must be computed (in the YYYY-MM-DD format); defaults to the current date
+    The date for which the ephemerides must be computed, either in the YYYY-MM-DD format or as an interval in the [+-] YyMmDd format; defaults to the current date
```
Comment (Deuchnord) [inline manpage/kosmorro.1.md:26]:
```suggestion
    The date for which the ephemerides must be computed, either in the YYYY-MM-DD format or as an interval in the "[+-]YyMmDd" format (with Y, M and D numbers); defaults to the current date
```

File: kosmorrolib/main.py (hunk 1 above)
Comment (Brainface1) [reply kosmorrolib/main.py:103]: Yeah that would be nice, but the functions output currently changes with the current date. So we would have to add something like `current_date` to the parameters of the function.

File: manpage/kosmorro.1.md (hunk 2 above)
Comment (Brainface1) [reply manpage/kosmorro.1.md:26]: This clarifies it a lot

**Comment by Brainface1** [comment]
And i found the problem with the dateutil package.
Its called python-dateutil

File: kosmorrolib/main.py (hunk 1 above)
Comment (Deuchnord) [reply kosmorrolib/main.py:103]: Actually, you can use the same `relativedelta()` function in your test to check you get the same date as expected, no matter what the current date is. The goal is just to check the string is interpreted as expected :)

**Comment by Brainface1** [comment]
You want everything in one commit right? So how should i go about implementing these changes?

File: kosmorrolib/main.py (hunk 1 above)
Comment (Brainface1) [reply kosmorrolib/main.py:103]: Ah ok

**Comment by Deuchnord** [comment]
Yes, I'd prefer having everything in one single commit.
Just edit your file as usual, then commit with the `--amend` argument and push with `--force`.

**Comment by Brainface1** [comment]
Okay so the Pipfile + Pipfile.lock are now up-to date and installable and the unit tests run just fine in the CI, but in the E2E tests the python-dateutil package still cant be imported and there seems to be a problem with the locales.

**Comment by Deuchnord** [comment]
> in the E2E tests the python-dateutil package still cant be imported

Sorry, I forgot that the `setup.py` had to be updated too: on line 44, you have to add `python-dateutil` too, because it's used to package Kosmorro.
I'll have to find a way to simplify this :thinking: 

> there seems to be a problem with the locales.

The job tells you what's happening: in order to update the commands, invoke the following command:
```bash
pipenv run python setup.py extract_messages --output-file=kosmorrolib/locales/messages.pot
```
then commit the `kosmorrolib/locales/messages.pot` file :)

**Comment by Brainface1** [comment]
Now finally everything is working and the checks were successful. Thanks so much, you were incredibly helpful and i have learned a lot. (Reading the CONTRIBUTING.md page carfully for example)

File: test/core.py [hunk 3]
```diff
@@ -27,6 +30,10 @@ def test_get_env(self):
 
         self.assertEqual("{'great_variable': 'value', 'another_variable': 'another value'}", str(env))
 
+    def test_date_arg_parsing(self):
+        self.assertEqual(core.get_date("+1y 2m3d"), date.today() + relativedelta(years=1, months=2, days=3))
+        self.assertEqual(core.get_date("-1y2d"), date.today() - relativedelta(years=1, days=2))
+        self.assertEqual(core.get_date("1111-11-11"), date(1111, 11, 11))
```
Comment (Deuchnord) [inline test/core.py:36]:
```suggestion
        self.assertEqual(core.get_date("1111-11-13"), date(1111, 11, 13))
```
This way, we make sure the month and the day are correctly parsed (if they happen to be inverted, the date will become invalid and the CI will fail)

**Comment by Deuchnord** [comment]
Thanks @Brainface1!

**Pull request merged**